src/libs/ReportUtils.js

```diff
@@ -964,6 +964,24 @@ function shouldReportBeInOptionList(report, reportIDFromRoute, isInGSDMode, curr
     return true;
 }
 
+/**
+ * Attempts to find a report in onyx with the provided list of participants
+ * @param {Array} newParticipantList
+ * @param {Array} reports
+ * @returns {Array|undefined}
+ */
+function getChatByParticipants(newParticipantList, reports) {
```

Comment:
I think that we should just connect ReportUtils to the reports Onyxkey instead of passing it as an argument. We will call this method from other places, e.g. IOU pages, and having to connect each of those individual pages to the report key seems unecessary.

Reply:
I'm super down for this, the reason I didn't do it initially is that I don't really know the performance impacts of connecting onyx to a utils file

Reply:
cc: @tgolen since you probably have the most info

Reply:
I think that's ok. ReportUtils already connects to several Onyx keys - https://github.com/Expensify/App/blob/b8cee2889ad8415f3ad9dae9a15c82ae2f122b0a/src/libs/ReportUtils.js#L19-L59 and several other utils files do the same.

Reply:
Yeah, to be honest, we still don't have very clear guidance on doing this. There are some theoretical problems that we can imagine happening, but are unable to prove, and would be difficult to diagnose if they did happen. It mostly relates to race conditions between when `withOnyx()` sends data to a view and when `Onyx.connect()` sends data to a connection callback.

So, adding it as a `.connect()` in the file, should be fine, and there isn't any issue with that which we have found.

On the flip side, passing the data down as arguments will ensure that there is never a race condition and the data is also exactly what you expect it to be.